### DISCUSSION TITLE
Modularized :: MQTT :: viseron/toggle/<CAMERA>_connection/command

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#321] [Q&A] [answered]
Noticed that the modularized version, MQTT uses `viseron/toggle/<CAMERA>_connection/command`

Also the thumbnail MQTT has changes - much better path

Are there changes in HA I need to make to utilize the new `viseron/toggle/<CAMERA>_connection/command` path?

And is `camera `available yet in MQTT? (Think I remember seeing that it wasn't there yet.)

This is awesome - enjoying the modularized journey!

**Comment by roflcoopter** [comment]
IF you use MQTT discovery you should have a switch entity available which uses the toggle topic.

It didnt make sense to have a switch type entity in Viseron so i went for a toggle instead, but that is then mapped to a switch entity in Home Assistant

**Comment by roflcoopter** [comment, marked as answer]
And no the cameras are not implemented yet, not sure if it makes sense to do that really.
Old Viseron pushed images over MQTT which consumed a lot of CPU, so i might drop that in favor of re-streaming instead, but that means you would have to configure Home Assistant to see them

**Reply by gniknalu** [reply]
> And no the cameras are not implemented yet, not sure if it makes sense to do that really. Old Viseron pushed images over MQTT which consumed a lot of CPU, so i might drop that in favor of re-streaming instead, but that means you would have to configure Home Assistant to see them

OK - sounds very reasonable. I'll put the stream URL into HA instead.

Thank you!
